### PR TITLE
Remove imports of `dart:io` from `test.dart` unless its available

### PR DESCRIPTION
**Pull request by jakemac53**
Adds conditional imports and moves some code around to eliminate non-conditional imports of `dart:io` from `test.dart`.

### CODE REVIEW

File: lib/src/util/io_stub.dart [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import '../backend/runtime.dart';
```
Comment (natebosch) [inline lib/src/util/io_stub.dart]: [nit] copyright?

File: lib/test.dart [hunk 2]
```diff
@@ -39,6 +38,8 @@ export 'src/frontend/throws_matchers.dart';
 export 'src/frontend/timeout.dart';
 export 'src/frontend/utils.dart';
 
+import 'package:path/path.dart' as p;
```
Comment (natebosch) [inline lib/test.dart]: why did this move?

Comment (jakemac53) [reply lib/test.dart]: just accidental I think, was playing around with a few things and we don't have the import order lint I guess

**Comment by jakemac53** [comment]
Running internal presubmit, assuming that is happy I will submit tomorrow

**Pull request merged**